### DISCUSSION TITLE
how to change/edit Il2Cpp.Object

### DISCUSSION THREAD

**Discussion by marioAI** [vfsfitvnm/frida-il2cpp-bridge#233] [Q&A]
I've tried to change some function arguments with an "Interceptor" but have no idea how to change the String inside the Il2Cpp.Object

```
const listener = Interceptor.attach(requestinfohandler.virtualAddress, {
   "onEnter": function (args) {`
      console.log("ON ENTER");
      var data = new Il2Cpp.Object(args[2]);  
      console.log(data.toString().replace('xxx', 'yyy'))
    },
    "onLeave": function (retval) {
      console.log("ON LEAVE");
    }

});
```
here I somehow need to change the information in the Il2cpp.Object and then "return" it to the args[2]

**Comment by bluewave41** [comment]
`object.field('field').value = Il2Cpp.String.from('new value');` will change a _field_ on an Il2Cpp.Object.

If args[2] itself is a string you would `return Il2Cpp.String.from(args[2].replace('xxx', 'yyy'))`

Il2CppObject is a class which is why you're using `new`. The implementation is at https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/structs/object.ts

**Reply by marioAI** [reply]
Ok, thanks.

args[2] is of type NativePointer, thats why I made a new Il2Cpp.Object out of it as a first try.
If I try to make a Il2Cpp.String form the Pointer I only get "";

Now I tried:
```
var data = new Il2Cpp.Object(args[2]);
data.field("_data").value = Il2Cpp.String.from(data.field("_data").value.toString().replace('xxx', 'yyy'));
args[2].writePointer(data.handle);
```
but that crashes the game.

**Reply by bluewave41** [reply]
I'm not at all familiar with raw frida and what it looks like you're doing isn't something I've ever had to do. The library handles all this through it's methods. Unfortunately all I can tell you is that what you did doesn't make sense.

Typically what you'd end up with using this library is something like:

```
const assem = Il2Cpp.Domain.tryAssembly('Assembly-CSharp').image;
assem.class('Thing').method('DoStuff').implementation = function(str) {
    return Il2Cpp.String.from(str.replace('xxx', 'yyy');
}
```

That will change the Thing.DoStuff method from the Assembly-CSharp assembly.

Maybe someone else can explain why your code doesn't work. Sorry!

**Reply by marioAI** [reply]
Ok, thanks again.

I tried the "implementation" stuff too, but failed at first because I could not invoke the function with "this".
But I could get it to run now, still have some troubles with the Il2Cpp.String object there.

(see discussion: [#234 ](https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/234) )
